WISE-Community/WISE-Client
# refactor(Grade By Student): reduce duplication and complexity

## Changes
- Use ng-template for th elements
- Replace long switch statement with object literal

## Test
- Make sure sorting and filtering still works as before in Grade by Student view
- Ensure that First Name and Last Name columns are only visible when teacher has permission to view names
- Ensure that Student column is only visible when teacher doesn't have permission to view names
